target/linux/ath79/dts/ar9344_devolo_magic-2-wifi.dts

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9344.dtsi"
+
+/ {
+	model = "Devolo Magic 2 Wifi";
+	compatible = "devolo,magic-2-wifi", "qca,ar9344";
+
+	aliases {
+		led-boot = &plcr;
+		led-failsafe = &plcr;
+		led-running = &plcw;
+		led-upgrade = &plcr;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan: wlan {
+			label = "devolo:status:wlan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		plcw: lan_status {
+			label = "devolo:status:dlan";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "netdev";
+		};
+
+		plcr: dlan_error {
+			label = "devolo:error:dlan";
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+			panic-indicator;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset_button {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+		wifi_button {
+			label = "WIFI button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 20 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+		dlan_button {
+			label = "DLAN button";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&gpio {
+	status = "okay";
+
+	wlan_power {
+		gpio-hog;
+		line-name = "WLAN power";
+		gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+
+	plc_enable {
+		line-name = "PLC enable";
+		gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+
+	plc_pairing {
+		line-name = "PLC pairing";
+		gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		output-high;
+	};
```

Comment:
I don't think you need to define plc_enable and plc_pairing here if you have it in 03_gpio_switches.
But I'm only 90 % sure.

Reply:
I definitely want them defined in the dts too with their appropriate label, so users without knowledge can locate it easily in the dts and can allow in the future better handling when it evolves

Reply:
It would be good if you removed the "resolved" here, as I also want to make this discussion aware to other reviewers.

Reply:
I think adding unused device tree nodes doesn't make much sense. If it evolves there will be tree-wide replacement of gpio_switches and we could simply use information in the script instead. And when this happens, these extra nodes may be overlooked by other authors.

Reply:
> It would be good if you removed the "resolved" here, as I also want to make this discussion aware to other reviewers.

Sorry @adrianschmutzler, I never got to see that comment before

Reply:
Is there a way to bind a button to this gpio in the dts ?
The idea is that while the user press that button, the kernel changes the gpio 11 to 0...

Reply:
While you could easily do so, OpenWrt policy is to keep LEDs/buttons to the use that has been defined it stock firmware. So, unless stock firmware uses a button for that, you will have to stick to the uci config solution (at least for the support to be merged in master).

Reply:
That is precisely the stock firmware behavior on this device :)
Is there a way to do it?

Reply:
What exactly is the stock firmware behavior?

Reply:
As I said, stock firmware behavior is gpio=1 while the button is pressed like this
`if [ "$ACTION" = "pressed" ]; then
        logger "press pairing button"
        echo 1 > $PLC_BUTTON
elif [ "$ACTION" = "released" ]; then
        logger "release pairing button"
        echo 0 > $PLC_BUTTON
fi`

Reply:
This is a patch where a custom button has been added. However, the button still seems to have been set to a predefined task:
https://github.com/openwrt/openwrt/commit/11182349e1f31f873ebddd69d6b87dec638eaabf
https://github.com/openwrt/openwrt/commit/4d5dae27419a6883484d0a051c653888d2042bca

Maybe you can work on from that.

Reply:
Removed these nodes from dts